code-reviews/recipes/Terraform.md

```diff
@@ -53,10 +58,16 @@ echo -e "\n\n>>> Terraform validate"
 terraform validate
 ```
 
-> TODO: Add example of how to use this in an AzDO build validation pipeline
-
-## Code Review Checklist
-
+### Code Review Checklist
 In addition to the [Code Review Checklist](../readme.md) you should also look for these terraform specific code review items
 
-> TODO: Add checklist
+- Verify Terraform project is configured to use Azure Storage as remote state backend
+- Verify remote state backend storage account key is stored in the key vault and not hard coded
+- If Terraform code is mixed with application source code, verify that the Terraform code is isolated into a dedicated folder
+- If the infrastructure going to different depending on the environment (Example: Dev, UAT, Production), verify sub-directory per environment is created
```

Comment:
Not sure I understand [agree with this]. You should deploy the same template into each environment. This isn't very DRY to have multiple env templates. Unique names for the state should help with this.

Reply:
@ianphil - In some scenarios, While UAT and Prod environments are exactly the same,  integration testing environments are provisioned with lower spec workloads. While the template itself will be same for all environments, the sku/size could be different. We may recommend to have environment specific tfvar that supplies environment specific values. thoughts?

Reply:
Resolving the conversation as there was no further feedback from @ianphil